### ISSUE TITLE
helm cache is keeping grow almost hit 30G and deleting it caused issue

### ISSUE THREAD

**Issue by xyfleet** [hashicorp/terraform-provider-helm#1015]
### Terraform, Provider, Kubernetes and Helm Versions

```
Terraform version: Terraform v1.3.6 on darwin_arm64
Provider version: hashicorp/helm v2.7.1
Kubernetes version: hashicorp/kubernetes v2.16.0 (kubernetes cluster: 1.22)
```

### Affected Resource(s)
helm_release.aws_ebs_csi_driver
helm_release.kiali_operator

### Terraform Configuration Files

List part of helm_release configuration.
```
resource "helm_release" "aws_ebs_csi_driver" {
  name       = "ebs-csi"
  chart      = "aws-ebs-csi-driver"
  repository = "https://kubernetes-sigs.github.io/aws-ebs-csi-driver"
  version    = ">=2.0.4, <2.6.0"
}

resource "helm_release" "kiali_operator" {
  name       = "kiali-operator"
  chart      = "kiali-operator"
  repository = "https://kiali.org/helm-charts/"
  version    = "v1.46.0"
}


```

### Expected Behavior
After deleting the helm cache files under `/Users/sam/Library/Caches/helm/repository`, and commit `terraform init, terraform plan`, should return Plan 0 add, 0 change, because there is nothing configuration changed. 

### Actual Behavior
But it showed me **Plan 0 add, 2 change.** Something like below:
```
  ~ resource "helm_release" "kiali_operator" {
        id                         = "kiali-operator"
        name                       = "kiali-operator"
      ~ version                    = "1.46.0" -> "v1.46.0" 

 # module.eks.helm_release.aws_ebs_csi_driver[0] will be updated in-place
  ~ resource "helm_release" "aws_ebs_csi_driver" {
        id                         = "ebs-csi"
        name                       = "ebs-csi"
      ~ version                    = "2.5.1" -> ">=2.0.4, <2.6.0"
        # (26 unchanged attributes hidden)
    }
```

### Another issue
I tried to upgrade helm_release version of ebs csi driver to "2.13.0" from ">=2.0.4, <2.6.0", but it failed, after deleting the cache files. See the error below. (Restoring the cache files back, the upgrade will not have any issue. I tested several times, I can reproduce this 100% )

```
╷
│ Error: cannot patch "ebs.csi.aws.com" with kind CSIDriver: CSIDriver.storage.k8s.io "ebs.csi.aws.com" is invalid: spec.fsGroupPolicy: Invalid value: "File": field is immutable
│
│   with module.eks.helm_release.aws_ebs_csi_driver[0],
│   on ../../modules/eks/main.tf line 203, in resource "helm_release" "aws_ebs_csi_driver":
│  203: resource "helm_release" "aws_ebs_csi_driver" {
```

These cache files get bigger and bigger, eventually eating up my disk. How can I handle this situation?

**Comment by xyfleet** [comment]
any update will be appreciated.

**Comment by arybolovlev** [comment]
Hi @xyfleet,

The problem with the cache that you encounter has been addressed in Helm `v3.10.1`([Helm release note](https://github.com/helm/helm/releases/tag/v3.10.1)), however, we haven't yet updated the provider's dependencies and still use `v3.9.4`. Once we update dependencies, this issue will be solved. As for now, the workaround will be to clean up the Helm cache manually. [Here](https://helm.sh/docs/helm/helm/) you can find more details about the Helm cache.

As you have mentioned, once you cleaned the cache, you get another issue when Terraform discovers changes and wants to perform an update. I was able to reproduce this issue and need some time to figure out if that can be addressed.

As for now, I can propose the following workarounds for this issue:
1. once you clean the cache, do not delete index files, but just temporary ones. Something like this(works on macOS and Linux): `rm *=-charts.txt` and `rm *=-index.yaml`;
2. in the case you have removed all files from the cache, including indexes, run `helm repo update` to re-fetch index files for your repositories, then Terraform will continue to perform normally.

I hope that helps for now and in the meantime, we will dig deeper into this issue.

Thank you!

**Comment by arybolovlev** [comment]
I will go ahead and close this issue since the Helm package update was merged.

The change will be available in the upcoming release. Please keep an eye on the release notes.

Thank you for reporting this issue.
